vowpalwabbit/parse_regressor.cc

```diff
@@ -349,8 +348,18 @@ void save_load_header(
         else  // < VERSION_FILE_WITH_INTERACTIONS
         {
           // pairs and triples may be restored but not reflected in interactions
-          all.interactions.insert(std::end(all.interactions), std::begin(all.pairs), std::end(all.pairs));
-          all.interactions.insert(std::end(all.interactions), std::begin(all.triples), std::end(all.triples));
+          std::vector<std::vector<namespace_index>> new_pairs(all.pairs.size());
+          std::vector<std::vector<namespace_index>> new_triples(all.triples.size());
+
+          for (auto &t : all.pairs){
+            new_pairs.emplace_back(t.begin(), t.end());
```

Comment:
Why can't you simply insert into  all.interactions directly from all.pairs and all.triples?

Reply:
all.interactions is changed to now have a default type `vector<vector<namespace_index>>` instead of `vector<string>`. So it requires me to create the temporary `new_pairs` and `new_triples` objects.

Reply:
`all.interactions.emplace_back(t.begin(), t.end());` ?

Reply:
Ah okay. I get it now. I'll change this.

Reply:
I get why we're using emplace_back here, but its really just hacking around the fact that pairs/triples is an incompatible type. How feasible would it be to convert all the interactions (including pairs/triples/interactions function signatures) to uint8s?